### PR TITLE
Stories

### LINKED ISSUE #1040: Story Publishing
- [ ] Differentiate between Official and un-official stories:
      - Official stories have an official Header, un-official/community stories do not have such a header
      - Header consists of: background image, title, subtitle (e.g. "official story by ESA-NASA-JAXA" , "featured story from EO Dashboard hackathon", label (?)
- [ ] check terms and conditions and amend as needed to deal with cases of inapropriate content being published in un-official stories (Anca)
- [ ] assign **publisher** = person in charge of story layout editing and publishing (Anca --- could be Sara)

- [ ] assess effort and opportunity for new Git submodule to manage story publishing
        - use Github for authentication
        - store Story IDs in json file 
        - configure json with 4-5 sections (=number of themes)
        - to add a new story edit the json by inputing the story metadata in the correct section in the json
        - published = story content is online and linked to the main page
- [ ] assess opportunity to implement "forking" approach to limit editing of community story post-publication 
        - publishing team forks community contribution and publishes
        - author notifies team when new version is ready 
        - publishing team re-forks and publishes and updates the json
        
- [ ] Workflow to publish stories:
      - add new indicator
      - science team defines story 
      - graphics team creates the header 
      - science + comm review the header
      - communication team edits the content
      - comm + science create the custom-dashboard 
      - publisher edits layout
      - comm + science approve final layout
      - all review on staging with header
      - publisher publishes 
      
- [ ] rebrand custom-dashboard to story
- [ ] keep track of the list of custom-views - list all dashboards in every new email

### LINKED ISSUE #1064: New landing page overlay
- overlay over all of current landing page (except hedaer/footer), closeable with a "x" icon
- some way to re-open it once it is closed (menu entry?)
- have multiple categories as tabs
- for each tab, have a grid/masonry-masonry style layout with the individual stories (image, title, subtitle)
- consider having a header image/carousel

see also #993

### LINKED ISSUE #1065: Header section for official custom dashboards
The header section of official custom dashboards/stories should include
- header image
- title
- subtitle
see also #1040

### LINKED ISSUE #1066: Create JSON/YAML-file / folder structure to contain selection of official stories
A hierarchical (categories!) JSON/YAML file includes the selection of dashboard IDs (custom dashboards), plus title, subtitle, relative image url. This feeds the new landing page overlay (#1064) with data.

Pseudocode:
```
{
  "Biomass": {
    "id": "a8843lsjd9",
    "title": "The first offical dashboard",
    "subtitle": "Created by ESA",
    "image": "/abc.jpg" or "https://race.esa.int/image.jpg"
  },
[...]
}
```

### LINKED ISSUE #1067: Create script to fetch and store official stories
This script should take the config file (#1066) and fetch the corresponding dashboards from the custom dashboard API, and store them into a subfolder. This script should be manually executable from the command line.

The script should probably also generate a new viewing UUID, to differenciate from the "still live" original dashboard (so the original one can still be updated).

### LINKED ISSUE #1068: Allow custom dashboard view to load local dashboards
Prior to loading custom dashboards from the dashboard api, the custom dashboard view should first check if the story with the given viewing ID is present in the locally saved dashboards folder (see #1067). If so, then this story is an "official one" (meaning the header is displayed, see #1065), and editing mode is never enabled for these.

### LINKED ISSUE #1069: Adapt custom dashboard layout to new "story" mode
Possible improvements:
- [ ] consider modifying the height of the rows to either take up more or less screen space; maybe it's best to take full screen width for a more "Immersive" look and feel
- [ ] scroll block maps and charts in custom dashboard to prevent accidental map/chart scrolling, maybe the user needs to press "ctrl" together with scroll to really scroll

### LINKED ISSUE #1079: Fix scroll on custom dashboard load
When initially loading a custom dashboard, sometimes the page scrolls down to ~half page height, when it actually should load the top of the page. Investigate what causes this and fix it.

### LINKED ISSUE #1217: Create landing pages for theme selection and individual themes
The theme selection landing page should
- display a short introduction to eodashboard
- allow the user to choose a specific theme
- show a selection of stories
- show an information about the "explore" mode (=map interface)

The individual theme landing page should
- be styled according to the color scheme of the current theme
- display a short introduction to the current theme
- display all stories for the current theme
- show an information about the "explore" mode (=map interface) which is filtered by the theme
- show a selection of the other available themes

### LINKED ISSUE #1244: Add a way to remove theme in the explore view
Currently, if filtered by a theme and in the explore view, the only way to see all POIs is to go back to the landing page, and then go back to explore view.

It would be nice to have a way to reset the theme (and showing all POIs) without leaving the explore view.

### PR DESCRIPTION
**Pull request by silvester-pari**
Adds stories functionality as described in #1040.

Closes #1040, closes #1064, closes #1065, closes #1066, closes #1067, closes #1068, closes #1069, closes #1079, closes #1217. closes #1244

This PR takes a bit different approach than originally intended, since it does not replace the `custom-dashboard` functionality, but rather adds the new `stories` functionality.

Additionally, it adds an "official" mode for custom dashboards, which allows setting header images and caching custom dashboards locally (to prevent them being automatically updated after publishing).

So, basically, features can be mixed and matched:
- "normal" custom dashboards (like before)
- "official" custom dashboards, with header image, subtitle, and consortium logos
- "normal" story, a dashboard viewing id used via the `/story` path, displaying the rows in full window height and adding scroll button
- "official" story, a cached custom dashboard with header, subtitle etc. plus "story" mode
I believe this way, we keep nice options and backwards compatibility, while adding the story mode on top of it.

Screenshots:
image
image
image
image

### CODE REVIEW

**Comment by santilland** [comment]
Some issues discovered for mobile view
- [ ] Story "start" page, if title to long, start button outside of screen, not possible to scroll
- [ ] Swiping pages in story currently not working (at least on my device)
- [ ] Text block ^ arrow symbol can't be seen in normal mode, only in dark mode

**Review by santilland — APPROVED** [review]
Looks good in principle, lets merge and see
